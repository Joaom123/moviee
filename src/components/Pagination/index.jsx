import React, { Fragment } from "react";
import "./pagination.css";

function Pagination({totalResults, page, handleChangeOfPage}) {
    const PageLink = (number) => {
        return (
            <li className={page === number ? "active" : "waves-effect"} key={number}>
                <a href="#!" onClick={(e) => handleChangeOfPage(e, number)}>{number}</a>
            </li>
        )
    }

    const PagesLinks = () => {
        const numberOfPages = Math.ceil(totalResults / 10);
        let pageElements = [];

        if (numberOfPages <= 10) {
            for (let i = 1; i <= numberOfPages; i++)
                pageElements.push(PageLink(i));

            return (
                <Fragment>
                    {pageElements}
                </Fragment>
            );
        }

        if (numberOfPages === page) {
            pageElements.push(PageLink(1));
            pageElements.push(<span key={numberOfPages + 1}>...</span>);

            for (let i = page - 5; i <= page; i++)
                pageElements.push(PageLink(i));

            return (
                <Fragment>
                    {pageElements}
                </Fragment>
            );
        }

        if (numberOfPages - page < 5) {
            for (let i = page; i <= numberOfPages; i++)
                pageElements.push(PageLink(i));

            return (
                <Fragment>
                    {pageElements}
                </Fragment>
            );
        }

        if (page >= 5){
            for (let i = page - 3; i <= page + 1; i++)
                pageElements.push(PageLink(i));

            pageElements.push(<span key={numberOfPages + 1}>...</span>);
            pageElements.push(PageLink(numberOfPages));

            return (
                <Fragment>
                    {pageElements}
                </Fragment>
            );
        }

        for (let i = 1; i <= 5; i++)
            pageElements.push(PageLink(i));

        pageElements.push(<span key={numberOfPages + 1}>...</span>);
        pageElements.push(PageLink(numberOfPages));

        return (
            <Fragment>
                {pageElements}
            </Fragment>
        );
    }

    if(totalResults === 0)
        return null;

    return (
        <ul className="pagination">
            <li className={page === 1 ? "disabled" : "waves-effect"}>
                <a href="#!" onClick={(e) => handleChangeOfPage(e, page - 1)}>
                    <i className="material-icons">chevron_left</i></a>
            </li>
            <PagesLinks />
            <li className={page === Math.floor(totalResults / 10) ? "disabled" : "waves-effect"}>
                <a href="#!" onClick={(e) => handleChangeOfPage(e, page + 1)}>
                    <i className="material-icons">chevron_right</i></a>
            </li>
        </ul>
    );
}

export default Pagination;