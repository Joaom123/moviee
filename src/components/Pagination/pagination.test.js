import React from "react";
import {cleanup, render} from "@testing-library/react";
import Pagination from "./index";

afterEach(cleanup);

describe("Pagination", () => {
    it("snapshot when total results equals 0", function () {
        const {asFragment} = render(<Pagination totalResults={0}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("snapshot when total results > 0", function () {
        const {asFragment} = render(<Pagination totalResults={340}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});