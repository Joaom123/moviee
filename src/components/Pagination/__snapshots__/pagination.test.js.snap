// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination snapshot when total results > 0 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        <i
          class="material-icons"
        >
          chevron_left
        </i>
      </a>
    </li>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        1
      </a>
    </li>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        2
      </a>
    </li>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        3
      </a>
    </li>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        4
      </a>
    </li>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        5
      </a>
    </li>
    <span>
      ...
    </span>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        34
      </a>
    </li>
    <li
      class="waves-effect"
    >
      <a
        href="#!"
      >
        <i
          class="material-icons"
        >
          chevron_right
        </i>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`Pagination snapshot when total results equals 0 1`] = `<DocumentFragment />`;
